extensions/wikia/Email/EmailController.class.php

```diff
@@ -396,6 +406,80 @@ protected function getSocialMessages() {
 		];
 	}
 
+	private function generateMobileApplicationBadges() {
+		$mobileApplicationsLinks = $this->generateMobileApplicationLinks();
+		$hasMobileApplicationBadges = count($mobileApplicationsLinks) > 0;
+		$badges = [
+			'hasMobileApplicationBadges' => $hasMobileApplicationBadges
+		];
+
+		if ( $hasMobileApplicationBadges ) {
+			if ( $mobileApplicationsLinks['android'] ) {
+				$badges['android'] = [
+					'link' => $mobileApplicationsLinks['android'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-android-badge' )->text() )
+				];
+			}
+			if ( $mobileApplicationsLinks['ios'] ) {
+				$badges['ios'] = [
+					'link' => $mobileApplicationsLinks['ios'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-ios-badge' )->text() )
+				];
+			}
+		}
+
+		return $badges;
+	}
+
+	private function generateMobileApplicationLinks() {
+		$result = [];
```

Comment:
How about we cache these results based on siteId? We send a lot of emails, about 12k an hour, but sometimes up to ~200k in an hour: https://kibana.wikia-inc.com/#/dashboard/elasticsearch/Wikia%20Email
<img width="1276" alt="screen shot 2016-09-23 at 4 08 18 pm" src="https://cloud.githubusercontent.com/assets/1657792/18788645/001c8844-81a8-11e6-893f-1e59162a66b4.png">

That would save us from having to make a request out to the mobile-application service for every email we send, and save us from having to parse the massive response it sends back. I don't think those release names should change, so we don't have to worry about invalidating data. So maybe cache for 24 hours?

Here's an example of using the MW memcache library: https://github.com/Wikia/app/blob/dev/extensions/wikia/Email/Controller/CategoryAddController.class.php#L86-L90